ocaml/xapi/xapi_pool.ml

```diff
@@ -3717,3 +3753,47 @@ let get_guest_secureboot_readiness ~__context ~self:_ =
       `ready_no_dbx
   | _, _, _, _ ->
       `not_ready
+
+let unpack_bundle ifd =
+  let result =
+    Tar_ext.unpack_tar_file
+      ~dir:!Xapi_globs.bundle_repository_dir
+      ~ifd
+      ~max_size_limit:!Xapi_globs.bundle_max_size_limit
+  in
+  match result with
+  | Ok () ->
+      ()
+  | Error e ->
+      raise
+        Api_errors.(
+          Server_error
+            (bundle_unpacked_failed, [Tar_ext.unpack_error_to_string e])
+        )
+
+let put_bundle_handler (req : Request.t) s _ =
```

Comment:
How are errors returned by this handler? I seems that `Api_error.Server_error` exceptions are being raised by code below, but not turned into specific HTTP error responses. I think that http_svr just turns them all into 500 responses. Is that what we want?

Reply:
The HTTP error code could be unified to http_400_badrequest.
The real error code and message could be put into the task result. So a client has to get the status of the task to get error code and error message.

The CLI server can get the error code and message from the task as well. And then ask the xe client to print to stdout.

Reply:
I updated this code:
For unpacking failure, it will set the HTTP error code to 400 (bad request), set the task (created by XC or CLI) to failure, and add the error message into the task's `error_info`. CLI will parse the task's `error_info` and print it to stdout. XC doesn't parse the HTTP response message, XC will also parse the task's `error_info`.
For other failures, e.g. repo related error, etc, it will keep the existing behavior to raise Api_error.Server_error.

Reply:
There is an example for unpacking a corrupted bundle file. When executing `xe` command, it will show the below result:
```
xe pool-sync-bundle filename=test_tar_ext_corrupted_file.tar
Failed to unpack bundle file.
err: File corrupted
```

Reply:
I think a user-visible error should avoid an abbreviation like `err` and use `error` or `reason`. Including the path to the corrupted file in the error message would further add value to it.

Reply:
I changed `err` to `error`. 
```
xe pool-sync-bundle filename=test_tar_ext_corrupted_file.tar
Failed to unpack bundle file.
error: File corrupted
```
But xapi can not get the path for the corrupted file. 
The file path is contained in `Tar.Header` which is parsed by `Tar_unix.Archive.with_next_file`. 
The `corrupted` error is raised during `Tar_unix.Archive.with_next_file`. In this time, the `Tar.Header` fails to be parsed so that xapi can not know the file path.